W-12203553-Add-sys-prop-to-REST-module-ref-CP (#168)
mulesoft/docs-apikit: modules/ROOT/pages/apikit-4-xml-reference.adoc

----

== APIkit for REST System Properties

[%header, cols="1a,2a, 3a, 4a"]
|===
| System Property | Default | Description | Since
| `mule.xml.expandExternalEntities` | false | Allows parser to expand XML external entities. The default value is `false` to prevent attacks. | 1.3.7
| `mule.xml.expandInternalEntities` | false | Allows parser to expand XML internal entities. The default value is `false` to prevent attacks. | 1.3.7
| `apikit.console.disabled` | false | Disables console. | 1.3.7
| `apikit.console.stream.amf.model` | true | Streams AMF parser model to the console. | 1.5.0
| `raml.json_schema.fail_on_warning` | false | Fails when there is a `warning` while validating a payload when the RAML parser is configured. | 1.1.15
| `yagi.json_duplicate_keys_detection` | true | Checks no duplicate JSON object field names are encountered when the RAML parser is configured. | 1.1.15
| `yagi.json_block_unsafe_polymorphic_base_types` | true | Blocks the use of unsafe base types when the RAML parser is configured. | 1.5.0
| `mule.apikit.parser` | AUTO | Overrides parser type. Possible values are:

* AMF

* RAML

* AUTO 
| 1.3.7

> | `apikit.multipart.size.limit` a| 268435456 | Limits the size of the multipart payload to be validated by throwing `InvalidFormParameterException` if the incoming multipart payload exceeds the limit to avoid stack overflow errors | 1.8.0

Reviewer: I think it's necessary to clarify that this is expressed in bytes. I also don't think that it's necessary to clarify that this is to avoid overflow errors, but it's up to you
```suggestion
| `apikit.multipart.size.limit` a| 268435456 | Limits the size of the multipart payload to be validated by throwing `InvalidFormParameterException` if the incoming multipart payload exceeds the limit (expressed in bytes) | 1.8.0
```

> | `fullDomain` | localhost | Overrides the console testing to URL. | 1.1.15

Reviewer: The sentence doesn't make sense to me
```suggestion
| `fullDomain` | localhost | Overrides the console URL. | 1.1.15
```

|`headerStrictValidation` |No | Validates that request headers are defined in RAML file or belong to a well-known set of headers
|`queryParamsStrictValidation` |No | Validates that request query parameters are defined in RAML file
|===

== APIkit for REST Namespace and Schema

----
xmlns:apikit="http://www.mulesoft.org/schema/mule/mule-apikit"
xsi:schemaLocation="http://www.mulesoft.org/schema/mule/mule-apikit http://www.mulesoft.org/schema/mule/mule-apikit/current/mule-apikit.xsd"
----

> == APIkit for REST System Property

Reviewer: Plural maybe?
```suggestion
== APIkit for REST System Properties
```

Author: You're right! Thanks!
